util_curl.go

```diff
@@ -31,7 +31,12 @@ func buildCurlRequest(req *http.Request, httpCookiejar http.CookieJar) (curl str
 
 	// 3. Generate curl body
 	if req.Body != nil {
-		buf, _ := io.ReadAll(req.Body)
+		// httpclient.Do method will read the entire body and make it empty
+		// thus using req.GetBody() instead of req.Body since req.Body
+		// is empty after Do method.
+		// body here is a copy of original body
+		body, _ := req.GetBody()
+		buf, _ := io.ReadAll(body)
```

Comment:
@tttturtle-russ Thanks for creating a PR. Last night, I discovered the branch `main` was missing curl middleware. It seems it happened when I merged the branch `v2` into branch `main`.

Branch `main`
https://github.com/go-resty/resty/blob/6d941ac6f1ed35557b6fb3fb41cf74beeb9be2f0/resty.go#L195-L196

Branch `v2`
https://github.com/go-resty/resty/blob/370d744bf0a478e71832d591061b30a9f91ef402/client.go#L1484-L1486